nemo/collections/asr/parts/preprocessing/features.py

```diff
@@ -71,13 +73,17 @@ def normalize_batch(x, seq_len, normalize_type):
             x_std[i] = x[i, :, : seq_len[i].item()].std()
         # make sure x_std is not zero
         x_std += CONSTANT
-        return (x - x_mean.view(-1, 1, 1)) / x_std.view(-1, 1, 1)
+        return (x - x_mean.view(-1, 1, 1)) / x_std.view(-1, 1, 1), x_mean, x_std
```

Comment:
Returning three things here breaks downstream models such as WaveformFeaturizer and MelSpecprocessors. Is this necessary? If it is, have you checked that rest of the downstream classes continue to work?

Potentially, accept a new keyword arg such that if set, it returns 3 values otherwise it defaults to old 1 values.

Reply:
stream_step in the last comment works for all cases including ctc, rnnt, chunk-aware, regular, etc.
I can rename it to cache_aware_stream_step() but we don't have any other streaming step currently supported. If we add a new one later, we can use the same method and control it with a parameter or config.

Reply:
Please rename to cache_aware_stream_step(). It is not a good default to name it very generic thing.